analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpAssignmentFinder.cs

```diff
@@ -38,10 +38,18 @@ protected override bool IsAssignmentToIdentifier(SyntaxNode node, string identif
                 node = statement.Expression;
             }
             if ((anyAssignmentKind || node.IsKind(SyntaxKind.SimpleAssignmentExpression))
-                && node is AssignmentExpressionSyntax assignment
-                && assignment.Left.NameIs(identifierName))
+                && node switch
+                {
+                    AssignmentExpressionSyntax assignment
+                        when assignment.MapAssignmentArguments().FirstOrDefault(x => x.Left.NameIs(identifierName)) is { Right: { } right } => right,
+                    PostfixUnaryExpressionSyntax { Operand: { } operand } when operand.NameIs(identifierName) => operand,
+                    PrefixUnaryExpressionSyntax { Operand: { } operand } when operand.NameIs(identifierName) => operand,
```

Comment:
As for `PostfixUnaryExpressionSyntax`, I am not sure that we should consider in `IsAssignmentToIdentifier` the following cases:
- `SyntaxKind.UnaryPlusExpression`
- `SyntaxKind.UnaryMinusExpression`
- `SyntaxKind.BitwiseNotExpression`
- `SyntaxKind.LogicalNotExpression`
- `SyntaxKind.AddressOfExpression`
- `SyntaxKind.PointerIndirectionExpression`
- `SyntaxKind.IndexExpression`

The one I would definitely consider are:
- `SyntaxKind.PreIncrementExpression`
- `SyntaxKind.PreDecrementExpression`